### ISSUE TITLE
spring-cloud-config-client: Load properties from Vault first.

### ISSUE THREAD

**Issue by kis024** [spring-cloud/spring-cloud-config#2326]
We've added basic auth to our config server, we want to store credentials for config clients in vault.
This flow doesn't work since RestTemplate that connects to config server is constructed much earlier than properties are loaded from vault (spring-cloud-dependencies: 2022.0.4).

Possible solution: 
If spring-cloud-starter-vault-config is present in the class path, the following order of values resolution could be used:
- fetch values from vault
- resolve current property placeholders
- load other remote property sources
- resolve placeholders other remote property sources

**Comment by ryanjbaxter** [comment]
Are you using `spring.config.import`?  can you provide its configuration?

**Comment by kis024** [comment]
yes, `spring.config.import: vault://, configserver:https://...` or `spring.config.import: configserver:https://..., vault://`

**Comment by ryanjbaxter** [comment]
Are this in you application yaml/properties?

can you try setting an environment variable/system property just for the vault import and leave the config server import in the application configuration file?

**Comment by kis024** [comment]
Tried, same result but different behaviour (I was even surprised that it worked, I though spring.config.import from env var would override the value from config file :))
I noticed that connection to vault now is done before connecting to config-server.
But property values are still not resolved and authentication failed.

**Comment by ryanjbaxter** [comment]
Ok.  We will have to take a look at it, it might be a limitation of spring boot but we will look into it

**Comment by ENate** [comment]
Hi. Has this issue been resolved? I am trying to load config properties from spring cloud config server (using a username and password) after loading other spring config server from vault. It seems the property sources cannot be loaded by clients when I used ```spring.config.import: optional:configserver: http://username:password@config-server:PORT```. Any help will be welcomed.

**Comment by ryanjbaxter** [comment]
@ENate are you trying to loading the username and password to use to authenticate with the config server from vault?

Something like `spring.import.config=http://${config.server.username}:${config.server.password}@locahost:8888` where `config.server.username` and `config.server.password` are coming from vault?

**Comment by ENate** [comment]
Yes @ryanjbaxter thanks for putting it so clear. I have been looking on ways to handle this issue but cant find any. I did it this way (with a yaml file which is same basically) :
```
spring:
    import:
       config: http://${CONFIG_USER}:${CONFIG_PASSWORD}@localhost:8888
```
I even tried defining ```spring.cloud.config.username```and ```spring.cloud.config.password``` as keys inside vault so as to later pass them in place of username and password but this did not work. Thanks again
